### DISCUSSION TITLE
Allow individual colouring to status bar like lines

### DISCUSSION THREAD

**Discussion by whitney-cumber** [neomutt/neomutt#4524] [Development]
## What would you like to be able to do?

Currently the same colour formatting that is applied to `$status_format` is also applied to `$pager_format` (and potentially others, compose menu? browser?)

I'd like to be able to colour `$pager_format` (and all other status line like things) independent of `$status_format`.
Here is why:

Assuming almost the default values:
```
# This has a space after the colons, i.e. "Msgs: 12" instead of "Msgs:12"
set status_format = "-%r-NeoMutt: %D [Msgs: %?M?%M/?%m%?n? New: %n?%?o? Old: %o?%?d? Del: %d?%?F? Flag: %F?%?t? Tag: %t?%?p? Post: %p?%?b? Inc: %b?%?l? %l?]---(%?T?%T/?%s/%S)-%>-(%P)---"
set pager_format = "-%Z- %C/%m: %-20.20n   %s%*  -- (%P)"
```

Consider this colouring:
```
color status red black 'New: [0-9]+(/[0-9]+)?'
```

Now you receive an email titled: "New: 99 ways of using Neomutt"

Now the *pager* status line, which shows the subject, has the "New: 99" coloured but nothing else.  This is surprising/confusing.  (Since the subject is not under your control you cannot prevent that.)
This is just one example on how a regex intended for one status bar surfaces on another wrecking havoc with the colours.

Since the purpose (and thus almost all content) of the different status lines are distinct (mine have no overlap at all), they (pager, browser, ...) should be coloured separately.

## Considered Alternatives

I considered surrounding the individual status/pager elements with "weird" symbols to make the pattern match on a subject more unlikely.  This is only a mitigation but does not fix the problem.

## Additional Info

Similar to:

https://github.com/neomutt/neomutt/issues/3680 (colouring headers)
https://github.com/neomutt/neomutt/issues/3672 (status_format caveats)
https://github.com/neomutt/neomutt/issues/3660 (allow different colours in attachment menu)
https://github.com/neomutt/neomutt/issues/3617 (allow different colours in sidebar)

**Comment by auouymous** [comment]
Are you asking for separate `color pager_status` and `color status`, or something like `color index_subject` with a separate color object for each field in each status bar?

It would be nice to have separate color objects for every field, but right now only a fraction of the index fields even have them. I don't know how feasible it is to add per-field color objects. However, being able to colorize the other characters in a status bar's format is equally important, and a separate color object for each status bar would be a good start.

**Comment by whitney-cumber** [comment]
> Are you asking for separate color pager_status and color status, or something like color index_subject with a separate color object for each field in each status bar?

The former.  The latter is nice to have, too, but IMHO it does not scale well.  What if in addition to a field you also want the an intro-label to be coloured, e.g. "New: %n" or a space between two fields (becomes visible if used with `reverse` colour attribute), ... what is the smallest "field" (cf. %Z in $index_format), ...

See https://github.com/neomutt/neomutt/issues/3617 for an idea of mine about a syntax covering all colouring cases I can think of in all format stuffs.
